components/brave_wallet_ui/common/hooks/use-multi-chain-sell-assets.ts

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+
+// Utils
+import { getNetworkInfo } from '../../utils/network-utils'
+import {
+  getRampAssetSymbol
+} from '../../utils/asset-utils'
+
+// Types
+import {
+  BraveWallet,
+  SupportedOffRampNetworks
+} from '../../constants/types'
+import { WalletSelectors } from '../selectors'
+
+// Hooks
+import { useIsMounted } from './useIsMounted'
+import { useLib } from './useLib'
+import { useUnsafeWalletSelector } from './use-safe-selector'
+
+export const useMultiChainSellAssets = () => {
+  // Redux
+  const networkList = useUnsafeWalletSelector(WalletSelectors.networkList)
+  const defaultCurrencies = useUnsafeWalletSelector(WalletSelectors.defaultCurrencies)
+
+  // Hooks
+  const isMounted = useIsMounted()
+  const { getAllSellAssets, getSellAssetUrl } = useLib()
+
+  // State
+  const [sellAmount, setSellAmount] = React.useState<string>('')
+  const [selectedSellAsset, setSelectedSellAsset] = React.useState<BraveWallet.BlockchainToken | undefined>()
+  const [showSellModal, setShowSellModal] = React.useState<boolean>(false)
+  const [options, setOptions] = React.useState<
+    {
+      rampAssetOptions: BraveWallet.BlockchainToken[]
+      allAssetOptions: BraveWallet.BlockchainToken[]
+    }
+  >({
+    rampAssetOptions: [],
+    allAssetOptions: []
+  })
+
+  // Memos
+  const supportedSellAssetNetworks = React.useMemo((): BraveWallet.NetworkInfo[] => {
+    return networkList.filter(n =>
+      SupportedOffRampNetworks.includes(n.chainId)
+    )
+  }, [networkList])
+
+  const selectedSellAssetNetwork = React.useMemo((): BraveWallet.NetworkInfo | undefined => {
+    if (selectedSellAsset?.chainId && selectedSellAsset?.coin) {
+      return getNetworkInfo(selectedSellAsset.chainId, selectedSellAsset.coin, supportedSellAssetNetworks)
+    }
+    return undefined
+  }, [selectedSellAsset?.chainId, selectedSellAsset?.coin, supportedSellAssetNetworks])
+
+  // Methods
+  const getAllSellAssetOptions = React.useCallback(() => {
+    getAllSellAssets()
```

Comment:
You can catch the error for easier debugging incase something goes wrong in the request

Reply:
Added a catch!